enhancements/sig-policy/74-order-policy-execution/README.md

```diff
@@ -0,0 +1,209 @@
+can have to create duplicate policies and more time consuming if users have to manually
+manage policy processing order. If a user elects to group the templates in one policy,
+there is no guarantee the templates will be processed in order, which could leave
+the policy in a bad state if a required resource has not yet been processed. Allowing
+users to specify policy dependencies will alleviate these issues.
+
+### Goals
+
+1. Policy template dependencies: allow users to automatically process policy templates
+in a desired order and guarantee that this order will be stable
+2. Opt-in: any extra fields that are used to specify policy ordering should be optional,
+and policies that do not contain these fields should function as they did before in order
+to avoid issues with older policies
+
+### Non-Goals
+
+1. Status message matching: we plan to only allow users to conditionally process policy
+templates based on compliance state, so users will not be able to specify that they want a
+policy to be processed if another policy 
+
+## Proposal
+
+We would add an optional new `dependencies` field to the spec on the policy level as
+well as an `extraDependencies` field on the template level to allow users to specify
+conditions for policy processing on both the policy and template level. When these
+fields are set and their conditions are not met, the policy templates will not be
+picked up by template-sync and the affected policy templates will enter a "pending"
+state. When the conditions are met or the fields are not set, the policy templates
+will be processed normally.
+
+### User Stories
+
+1. As an author of a ConfigurationPolicy, I don’t want resources to be wasted when
+I know one object depends on another. If an object cannot be cleanly applied before
+another is created, I want to specify this in the ConfigurationPolicy, and not waste
+resources (CPU, k8s API requests) in the policy controllers.
+2. As an author of a ConfigurationPolicy, I want to specify the order resources are
+applied in a policy when the API is not declarative. For some APIs (eg OLM), if
+resources are applied in the wrong order they can end up in an unhealthy state
+that I want to avoid.
+3. As a user of PolicySets, I want policies inside a set to be applied in a
+certain order.
+4. As an author of many ConfigurationPolicies, I have several policies that
+depend on another policy. I want to be able to reference completely separate
+policies (outside of the Policy “wrapper”) as required before a configuration
+is applied. (Maybe allow a policy to reference the status of a policy that runs
+before it (success or failure) before it tries to process it. Allowing you to chain
+them together)
+5. As a cluster administrator, I want to do a declarative action on a managed
+cluster when a policy becomes non-compliant. So I want to “activate” a
+ConfigurationPolicy based on the state of another policy in the cluster.
+
```

Comment:
Can we call out a user story for the generator so we don't forget to add support there?